src/iface/ethernet.rs

```diff
@@ -637,3 +637,168 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::Packet;
+    use phy::Loopback;
+    use wire::{ArpRepr, ArpOperation};
+    use wire::{EthernetAddress, EthernetFrame};
+    use wire::{IpAddress, IpCidr, Ipv4Address};
+    use iface::{ArpCache, SliceArpCache, EthernetInterface};
+    use socket::SocketSet;
+
+    #[test]
+    fn no_icmp_to_broadcast() {
+        // Unknown Ipv4 Protocol
+        //
+        // Because the destination is the broadcast address
+        // this should not trigger and Destination Unreachable
+        // response. See RFC 1122 § 3.2.2.
+        let bytes = vec![
+            // Ethernet Frame
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0x52, 0x54, 0x00, 0x00, 0x00, 0x00,
+            0x08, 0x00,
+            // Ipv4 Header
+            0x45, 0x00, 0x00, 0x14,
+            0x00, 0x00, 0x00, 0x00,
+            0x40, 0x0c, 0xfb, 0xdd,
+            0x7f, 0x00, 0x00, 0x01,
+            0xff, 0xff, 0xff, 0xff,
+        ];
+
+        // Create a basic device
```

Comment:
This common initialization code can be taken out of the tests and into a setup function.

Reply:
Good point